pandas/core/series.py

```diff
@@ -1782,12 +1782,16 @@ def sort_index(self, axis=0, level=None, ascending=True, inplace=False,
                                                  sort_remaining=sort_remaining)
         elif isinstance(index, MultiIndex):
             from pandas.core.groupby import _lexsort_indexer
-            indexer = _lexsort_indexer(index.labels, orders=ascending)
-            indexer = _ensure_platform_int(indexer)
-            new_index = index.take(indexer)
+            indexer = _lexsort_indexer(index.labels, orders=ascending,
+                                       na_position=na_position)
```

Comment:
Yes, that is true.  Unfortunately, it has no effect right now.  I pushed that change now.